prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm.c

```diff
@@ -546,6 +557,13 @@ static inline int fi_ibv_rdm_tagged_poll_recv(struct fi_ibv_rdm_ep *ep)
 				struct fi_ibv_rdm_tagged_conn *conn =
 				    (struct fi_ibv_rdm_tagged_conn *)(uintptr_t)
 				    wc[i].wr_id;
+#if defined(__ICC) || defined(__INTEL_COMPILER) || \
+    defined(__GNUC__) || defined(__GNUG__)
+				_mm_prefetch((const char *)
+					     fi_ibv_rdm_tagged_get_rbuf(conn, ep, wc[i].imm_data),
+					     _MM_HINT_T0);
+#endif /* ICC || GCC */
+
```

Comment:
if-defs in C files results in code that is difficult to read and maintain.  If possible, please try to place the definitions in header files, such that you just get a no-op when not defined.  E.g. in the .h do:

if defined blah blah
   define PREFETCH(x,y)  __mm_prefetch(x,y)
else
  define PREFETCH(x,y)
endif